File: client/java/src/avt/StringObj.java
package avt;

public final class StringObj extends SubObject {
   public String str;
   public String str2;
   public int w2;
   private int dir = 1;
   public int dis = 0;
   public int anthor = 0;

   public StringObj() {
   }

   public StringObj(String var1, int var2) {
      this.str = var1;
      this.w2 = var2;
   }

   public StringObj(int var1, int var2, String var3) {
      super.x = var1;
      super.y = var2;
      this.str = var3;
   }

   public final void transTextLimit(int var1) {
      this.dis += this.dir;
      if (this.dis > this.w2 - var1) {
         this.dis = -20;
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/StringObj.cs
public class StringObj : SubObject
{
	public string str;

	public string str2;

	public int w2;

	public int dir = 1;

	public int dis;

	public int anthor;

	public StringObj()
	{
	}

	public StringObj(string st, int ww)
	{
		str = st;
		w2 = ww;
	}

	public StringObj(int x, int y, string str)
	{
		base.x = x;
		base.y = y;
		this.str = str;
	}

	public void reset()
	{
		dir = 1;
		dis = 0;
	}

	public void transTextLimit(int limit)
	{
		dis += dir;
		if (dis > w2 - limit)
		{
			dis = -20;
		}
	}
}
